core/os/semaphore.h

```diff
@@ -30,13 +30,13 @@
 
 #pragma once
 
-#ifdef THREADS_ENABLED
-
 #include "core/typedefs.h"
 #ifdef DEBUG_ENABLED
 #include "core/error/error_macros.h"
 #endif
 
+#ifdef THREADS_ENABLED
```

Comment:
```suggestion

#ifdef THREADS_ENABLED
#ifdef DEBUG_ENABLED
#include "core/error/error_macros.h"
#endif
```
IMO, unnecessary to include error macros that aren't used here